runtime/compiler/env/VMJ9.cpp

```diff
@@ -2460,7 +2460,10 @@ TR_J9VMBase::getClassSignature_DEPRECATED(TR_OpaqueClassBlock * clazz, int32_t &
    for (i = 0; i < numDims; i++)
       sig[i] = '[';
    if (* name != '[')
-      sig[i++] = 'L';
+      if (isValueTypeClass(myClass) || hasUnflattenedFlattenableFields(myClass))
```

Comment:
`Q` should be used only for a value type class - not for ordinary identity classes that have value type fields.  This should only check `isValueTypeClass`.